Tests/SwiftLexicalLookupTest/NameLookupTests.swift

```diff
@@ -835,4 +835,137 @@ final class testNameLookup: XCTestCase {
       ]
     )
   }
+
+  func testSwitchExpression() {
+    assertLexicalNameLookup(
+      source: """
+        switch {
+        case .x(let 1️⃣a, let 2️⃣b), .y(.c(let 3️⃣c), .z):
+          print(4️⃣a, 5️⃣b, 6️⃣c)
+        case .z(let 7️⃣a), .smth(let 8️⃣a)
+          print(9️⃣a)
+        default:
+          print(0️⃣a)
+        }
+        """,
+      references: [
+        "4️⃣": [.fromScope(SwitchCaseSyntax.self, expectedNames: ["1️⃣"])],
+        "5️⃣": [.fromScope(SwitchCaseSyntax.self, expectedNames: ["2️⃣"])],
+        "6️⃣": [.fromScope(SwitchCaseSyntax.self, expectedNames: ["3️⃣"])],
+        "9️⃣": [.fromScope(SwitchCaseSyntax.self, expectedNames: ["7️⃣", "8️⃣"])],
+        "0️⃣": [],
+      ],
+      expectedResultTypes: .all(IdentifierPatternSyntax.self)
+    )
+  }
+
+  func testSimpleGenericParameterScope() {
+    assertLexicalNameLookup(
+      source: """
+        class A<1️⃣T1, 2️⃣T2> {
+          let 7️⃣x: 3️⃣T1 = v
+          let y: 4️⃣T2 = v
+
+          class B<5️⃣T1> {
```

Comment:
I am curious what the compiler (and this implementation) would do for the second `T2` in `class B<T1, T2: T2>`.

Reply:
At least when it comes to this implementation, the result [would be empty](https://github.com/swiftlang/swift-syntax/blob/57183fc995200d6ed3a6cd8755a9c761883064fd/Tests/SwiftLexicalLookupTest/NameLookupTests.swift#L897).

Reply:
So, it seems that there's no ordering amongst the various generic parameters. Given this:
```swift
class A<X> {
  class B<T2: T1, T1> { }
}  
```

unqualified lookup is finding `T1` and the type checker is rejecting it:

```
1 | class A<X> {
2 |   class B<T1, T2: T1> { }
  |                   `- error: type 'T2' constrained to non-protocol, non-class type 'T1'
3 | }  
4 | 
```

Reply:
FWIW, here's a case where it's also semantically well-formed to refer to generic parameters out-of-order:

```
struct X<C: Collection<T>, T> { }
```